src/main/telemetry/hott.c

```diff
@@ -415,8 +444,88 @@ static void hottPrepareMessages(void) {
 #endif
 }
 
+#if defined (USE_HOTT_TEXTMODE) && defined (USE_CMS)
+static void hottTextmodeStart()
+{
+    // Increase menu speed
+    cfTaskInfo_t taskInfo;
+    getTaskInfo(TASK_TELEMETRY, &taskInfo);
+    telemetryPeriod = taskInfo.desiredPeriod;
+    rescheduleTask(TASK_TELEMETRY, TASK_PERIOD_HZ(1000));
+
+    rxSchedule = HOTT_TEXTMODE_RX_SCHEDULE;
+    txDelayUs = HOTT_TEXTMODE_TX_DELAY_US;
+}
+
+static void hottTextmodeStop()
+{
+    // Set back to avoid slow down of the FC
+    if (telemetryPeriod > 0)
+        rescheduleTask(TASK_TELEMETRY, telemetryPeriod);
+
+    rxSchedule = HOTT_RX_SCHEDULE;
+    txDelayUs = HOTT_TX_DELAY_US;
+}
+
+bool hottTextmodeIsAlive()
+{
+    return textmodeIsAlive;
+}
+
+void hottTextmodeGrab()
+{
+    hottTextModeMessage.esc = HOTT_EAM_SENSOR_TEXT_ID;
+}
+
+void hottTextmodeExit()
+{
+    hottTextModeMessage.esc = HOTT_TEXTMODE_ESC;
+}
+
+void hottTextmodeWriteChar(uint8_t column, uint8_t row, char c)
+{
+    if (column < HOTT_TEXTMODE_DISPLAY_COLUMNS && row < HOTT_TEXTMODE_DISPLAY_ROWS) {
+        if (hottTextModeMessage.txt[row][column] != c)
+            hottTextModeMessage.txt[row][column] = c;
+    }
+}
+
+static void processHottTextModeRequest(const uint8_t cmd)
+{
+    static bool setEscBack = false;
+
+    if (!textmodeIsAlive)
+    {
```

Comment:
Please use K&R style (braces on new line only for function definitions, on the same line otherwise).